### PR TITLE
Publish jars on release.

### LINKED ISSUE #5005: publish JAR to cloudrepo.io as part of the release process
Uploading the versions manually via the api isn't going to last. We should document how to configure auth for publishing to cloudrepo.io and use that in the `release_version` script.

### PR DESCRIPTION
**Pull request by davinchia**
## What
Closes airbytehq/airbyte#5005 .

## How
- Modify gradle script to add publishing task to all subprojects that we want to publish.
- Modify release script to publish jars on release.

## Recommended reading order
1. `build.gradle`

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -290,3 +290,49 @@ tasks.withType(AbstractArchiveTask) {
...
+                }
+            }
+
+            maven {
+                credentials {
+                    name 'cloudrepo'
+                    username System.getenv('CLOUDREPO_USER')
+                    password System.getenv('CLOUDREPO_PASSWORD')
+                }
+                url 'https://airbyte.mycloudrepo.io/repositories/airbyte-public-jars'
```
Comment (ChristopheDuong) [inline build.gradle:334]: would be cool to automatically bump version in the cloud git repo too

File: airbyte-server/build.gradle [hunk 2]
```diff
@@ -1,28 +1,5 @@
...
-}
-
-publishing {
-    publications {
-        shadow(MavenPublication) { publication ->
-            project.shadow.component(publication)
-        }
-    }
-    repositories {
-        mavenLocal()
```
Comment (ChristopheDuong) [inline airbyte-server/build.gradle:24]: we still need to publish locally for dev/testing

**Comment by davinchia** [comment]
Timestamps show that this was published 5 mins ago
```
➜  ~ cd ~/.m2/repository/io/airbyte/airbyte-analytics/0.28.1-alpha
(⎈ |docker-desktop:default)
➜  0.28.1-alpha ls -al
total 8
drwxr-xr-x  3 davinchia  staff   96 Jul 28 18:21 .
drwxr-xr-x  4 davinchia  staff  128 Jul 28 18:21 ..
-rw-r--r--  1 davinchia  staff  436 Jul 28 18:22 airbyte-analytics-0.28.1-alpha.pom
(⎈ |docker-desktop:default)
➜  0.28.1-alpha ls
airbyte-analytics-0.28.1-alpha.pom
```

File: airbyte-server/build.gradle (hunk 2 above)
Comment (davinchia) [reply airbyte-server/build.gradle:24]: should be ok now

**Review by ChristopheDuong — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (ChristopheDuong) [reply build.gradle:332]: Created an issue: https://github.com/airbytehq/airbyte/issues/5047

**Comment by davinchia** [comment]
follow tickets:
- https://github.com/airbytehq/airbyte/issues/5048
- https://github.com/airbytehq/airbyte/issues/5047

**Comment by ChristopheDuong** [comment]
> Timestamps show that this was published 5 mins ago
> 
> ```
> ➜  ~ cd ~/.m2/repository/io/airbyte/airbyte-analytics/0.28.1-alpha
> (⎈ |docker-desktop:default)
> ➜  0.28.1-alpha ls -al
> total 8
> drwxr-xr-x  3 davinchia  staff   96 Jul 28 18:21 .
> drwxr-xr-x  4 davinchia  staff  128 Jul 28 18:21 ..
> -rw-r--r--  1 davinchia  staff  436 Jul 28 18:22 airbyte-analytics-0.28.1-alpha.pom
> (⎈ |docker-desktop:default)
> ➜  0.28.1-alpha ls
> airbyte-analytics-0.28.1-alpha.pom
> ```
But there are no jars being published there? just pom files?

**Pull request merged**